### ISSUE TITLE
cookie authentication

### ISSUE THREAD

**Issue by plebhash** [braidpool/braidpool#58]
once https://github.com/rust-bitcoin/rust-bitcoincore-rpc/issues/316 is unblocked, we need a follow-up PR to implement @mcelrath 's suggestion made on this comment: https://github.com/braidpool/braidpool/pull/54#issuecomment-1781212058
>
> So the standard way to authenticate to bitcoind is to use cookie authentication, which is the file ~/.bitcoin/.cookie and allows the local user to use bicoind without rpcuser and rpcpass. Looks like we can do this:
>
> https://github.com/rust-bitcoin/rust-bitcoincore-rpc/blob/master/client/src/client.rs#L198
>
> ... can you get cookie auth working by default (and optional rpcuser/rpcpass)?

**Comment by plebhash** [comment]
turns out this was just a silly mistake
